docs/user/preparing_a_pr.md

```diff
@@ -0,0 +1,80 @@
+
+# Preparing a Pull Request
+
+After adding a new feature or implementing a bug fix, you will probably want to
+merge your fix into the garage master branch and make it available to others.
+This page will guide you through the process of creating a pull request and
+merging your changes.
+
+## Precursors
+
+Assuming you read and followed the [git workflow](git_workflow.md) guide,
+you're almost ready to submit a PR. You should make sure your feature branch is
+rebased on to the latest version of `rlworkgroup/master` and that your
+pre-commit hooks pass, then run the test suite as shown below.
+
+### Passing the Test Suite
+
+Before submitting a PR, you should make sure that your changes do not cause any
+breakages in the garage codebase. We do this by running the test suite, which
+includes all the unit and integration tests that exist on the master branch,
+plus any new ones you added. You can run the test suite locally from the garage
+root directory by running
+
+```sh
+make test
```

Comment:
This will try to run all the tests, including mujoco ones. Since mujoco is an optional dependency and we don't assume that everybody has a license, without which this will fail. I think we need a different solution here.

Reply:
The CI attempts to do the same though. I can mention that this runs all tests, and that the mujoco ones will fail (be ignored?) if a valid license isn't found.

Reply:
Not anymore. The CI doesn't run mujoco and mujco_long test tags for pull requests from forks.